contentcuration/contentcuration/frontend/shared/i18n/index.js

```diff
@@ -129,13 +129,16 @@ class Translator {
    * @param {object} defaultMessages - an object mapping message ids to default messages.
    */
   constructor(nameSpace, defaultMessages) {
-    this.nameSpace = nameSpace;
-    this.defaultMessages = defaultMessages;
+    this._nameSpace = nameSpace;
+    this._defaultMessages = defaultMessages;
+    for (const key in defaultMessages) {
+      this[`${key}$`] = this.$tr.bind(this, key);
+    }
   }
   $tr(messageId, args) {
     return $trWrapper(
-      this.nameSpace,
-      this.defaultMessages,
+      this._nameSpace,
+      this._defaultMessages,
```

Comment:
Do any of these changes affect existing usages of the translator class?

Reply:
Yes they allow the introduction and use of "feature string files" in Studio as far as I can recall.

Reply:
(They were cherry-picked from Kolibri)